auto_dev/services/eject/index.py

```diff
@@ -0,0 +1,243 @@
+        packages_base / config.fork_id.author / "agents" / Path.cwd().name,
+    ]
+
+    for path in possible_paths:
+        if path.exists():
+            print(f"Cleaning up existing directory: {path}")
+            shutil.rmtree(path)
+
+    packages_base.mkdir(parents=True, exist_ok=True)
+
+
+def _run_packages_lock() -> None:
+    """Run the packages lock command."""
+    expect_script = """#!/usr/bin/expect -f
+set timeout -1
+spawn autonomy packages lock
+while {1} {
+    expect {
+        "Select package type (dev, third_party):" {
+            send "dev\\r"
+            exp_continue
+        }
+        eof {
+            break
+        }
+    }
+}
+"""
+    script_path = Path("lock_packages.exp")
+    script_path.write_text(expect_script)
+    script_path.chmod(0o755)
+
+    try:
+        subprocess.run(["./lock_packages.exp"], check=True)
+    finally:
+        script_path.unlink()
+
+
+def _handle_dependencies(config: EjectConfig, component_dir: Path, ejected: List[PublicId]) -> List[PublicId]:
+    """Handle dependency ejection."""
+    if config.skip_dependencies:
+        return ejected
+
+    deps = build_dependency_tree_for_component(component_dir, config.component_type)
+    for dep in deps:
+        for component_type in ComponentType:
+            component_path = config.base_path / "packages" / component_type.value / dep.author / dep.name
+            if not component_path.exists():
+                continue
+            if _eject_single_component(dep, config.fork_id.author, dep.name, component_type.value):
+                ejected.append(dep)
+            break
+
+    return ejected
+
+
+def eject_component(config: EjectConfig) -> List[PublicId]:
+    """
+    Eject a component and its dependents.
+
+    Args:
+        config: EjectConfig object containing ejection parameters
+
+    Returns:
+        List of ejected component IDs
+
+    Raises:
+        ValueError: If dependencies are skipped but not already ejected
+    """
+    try:
+        component_dir = Path.cwd() / config.component_type / config.public_id.name
+        deps = build_dependency_tree_for_component(component_dir, config.component_type)
+
+        _eject_dependencies(deps, config.fork_id.author, config.base_path)
+
+        if not _eject_single_component(
+            config.public_id, config.fork_id.author, config.fork_id.name, config.component_type
+        ):
+            return []
+
+        ejected = [config.public_id]
+        ejected = _handle_dependencies(config, component_dir, ejected)
+
+        agent_config_path = Path.cwd() / DEFAULT_AEA_CONFIG_FILE
+        _handle_agent_config(config, agent_config_path)
+        _cleanup_package_directories(config)
+
+        subprocess.run(
+            ["aea", "publish", "--push-missing", "--local"],
+            check=True,
+            capture_output=True,
+            text=True,
+        )
+
+        with change_dir(".."):
+            _run_packages_lock()
```

Comment:
No need for this, just get the str name of the expected component check if its in dev or thirdparty keys
If it is, then dismiss skip, 
if not, if its in vendor folder its thirdparty if its been ejected its dev.